pkg/adaptor/cloud/azure/manager.go

```diff
@@ -39,6 +39,7 @@ func (_ *Manager) LoadEnv() {
 	cloud.DefaultToEnv(&azurecfg.SubscriptionId, "AZURE_SUBSCRIPTION_ID", "")
 	cloud.DefaultToEnv(&azurecfg.Region, "AZURE_REGION", "")
 	cloud.DefaultToEnv(&azurecfg.ResourceGroupName, "AZURE_RESOURCE_GROUP", "")
+	cloud.DefaultToEnv(&azurecfg.UserResourceGroupName, "AZURE_USER_RESOURCE_GROUP", "")
```

Comment:
can you explain why it's needed?

Reply:
We need to pass an AZURE_USER_RESOURCE_GROUP because for some Azure solutions like ARO the user might need to create a separate resource group that seems not to be able to be queried via the Azure SDK. In addition, Azure also creates an internal resource group that can be queried by the SDK. This user RG is needed because some resources like Subnet ID are created under that RG